packages/core/test/unit/query-generator/add-constraint-query.test.ts

```diff
@@ -0,0 +1,420 @@
+        default: 'ALTER TABLE [myTable] ADD CONSTRAINT [check] CHECK ([age] >= 10)',
+        sqlite: notSupportedError,
+        'mysql snowflake': checkNotSupportedError,
+      });
+    });
+
+    it('generates a constraint snippet for a check constraint with an array of values', () => {
+      expectsql(() => queryGenerator.addConstraintQuery('myTable', { name: 'check', type: 'CHECK', fields: ['role'], where: { age: ['admin', 'user', 'guest'] } }), {
+        default: `ALTER TABLE [myTable] ADD CONSTRAINT [check] CHECK ([age] IN ('admin', 'user', 'guest'))`,
+        mssql: `ALTER TABLE [myTable] ADD CONSTRAINT [check] CHECK ([age] IN (N'admin', N'user', N'guest'))`,
+        sqlite: notSupportedError,
+        'mysql snowflake': checkNotSupportedError,
+      });
+    });
+
+    it('generates a constraint snippet for a check constraint', () => {
+      expectsql(() => queryGenerator.addConstraintQuery('myTable', { type: 'CHECK', fields: ['age'], where: { age: { [Op.gte]: 10 } } }), {
+        default: 'ALTER TABLE [myTable] ADD CONSTRAINT [myTable_age_ck] CHECK ([age] >= 10)',
+        sqlite: notSupportedError,
+        'mysql snowflake': checkNotSupportedError,
+      });
+    });
+
+    it('generates a constraint snippet for a check constraint for a model', () => {
+      const MyModel = sequelize.define('MyModel', {});
+
+      expectsql(() => queryGenerator.addConstraintQuery(MyModel, { type: 'CHECK', fields: ['age'], where: { age: { [Op.gte]: 10 } } }), {
+        default: 'ALTER TABLE [MyModels] ADD CONSTRAINT [MyModels_age_ck] CHECK ([age] >= 10)',
+        sqlite: notSupportedError,
+        'mysql snowflake': checkNotSupportedError,
+      });
+    });
+
+    it('generates a constraint snippet for a check constraint with schema', () => {
+      expectsql(() => queryGenerator.addConstraintQuery({ tableName: 'myTable', schema: 'mySchema' }, { type: 'CHECK', fields: ['age'], where: { age: { [Op.gte]: 10 } } }), {
+        default: 'ALTER TABLE [mySchema].[myTable] ADD CONSTRAINT [myTable_age_ck] CHECK ([age] >= 10)',
+        sqlite: notSupportedError,
+        'mysql snowflake': checkNotSupportedError,
+      });
+    });
+
+    it('generates a constraint snippet for a check constraint with default schema', () => {
+      expectsql(() => queryGenerator.addConstraintQuery({ tableName: 'myTable', schema: dialect.getDefaultSchema() }, { type: 'CHECK', fields: ['age'], where: { age: { [Op.gte]: 10 } } }), {
+        default: 'ALTER TABLE [myTable] ADD CONSTRAINT [myTable_age_ck] CHECK ([age] >= 10)',
+        sqlite: notSupportedError,
+        'mysql snowflake': checkNotSupportedError,
+      });
+    });
+
+    it('generates a constraint snippet for a check constraint with globally set schema', () => {
+      const sequelizeSchema = createSequelizeInstance({ schema: 'mySchema' });
+      const queryGeneratorSchema = sequelizeSchema.getQueryInterface().queryGenerator;
```

Comment:
```suggestion
      const queryGeneratorSchema = sequelizeSchema.queryGenerator;
```

A few other places too :)

Reply:
Done